Code/Robot/Robot/Robot.ino

```diff
@@ -77,49 +123,85 @@ void loop(JsonDocument& doc) {
-          state = eDriving;
+        } else if (doc.containsKey("g")) {
+          parseJsonIntoQueue(moveQueue, doc);
+          state = eMoving;
         }
         break;
-      case eDriving:
-        DEBUG_PRINTLN("JUST ENTERED DRIVING STATE");
-        if (moveQueue->empty()) {
-          DEBUG_PRINTLN("MOVEQUEUE EMPTY.");
-          state = eWaitingToStart;  // Go back to waiting state if queue is empty
-          DEBUG_PRINTLN("STATE SET TO WAITING TO START!");
-          break;
-        }
-        driving_logic(moveQueue);
+      case eMoving:
+        executeMoveSequence(moveQueue);
+        // Done executing moves
+        state = eWaitingToStart;
         break;
     }
   }
+  // -------------------------------------------------
 }
 
-void deserializeDKeyIntoQueue(std::queue<Move>* moveQueue, JsonDocument& doc) {
-  for (JsonObject obj : doc["d"].as<JsonArray>()) {
+// New JSON format.
+// Needs to be abbreviated to save bytes.
+// Suggest: x for type, d for direction, t for duration, f l s and b fir types.
+/*
+{
+  "g": [
+    {
+      "type": "freedrive",
+      "direction": 1,
+      "duration": 5000
+    },
+    {
+      "type": "linefollow",
+      "stopDistance": 100
+    },
+    {
+      "type": "scissor",
+      "direction": 1, // 1 or 0 for top and bottom
+    },
+    {
+      "type": "belt",
+      "duration": 2000
+    }
+  ]
+}
+*/
+void parseJsonIntoQueue(std::queue<Move>* moveQueue, JsonDocument& doc) {
+  for (JsonObject obj : doc["g"].as<JsonArray>()) {  // g for go
     Move currentMove;
-    currentMove.direction = obj["a"];
-    currentMove.time = obj["t"];
-    moveQueue->push(currentMove);
 
-    DEBUG_PRINT("Pushed Move - Direction: ");
-    DEBUG_PRINT(currentMove.direction);
-    DEBUG_PRINT(", Time: ");
-    DEBUG_PRINTLN(currentMove.time);
+    // Populate move structs params based on move type
+    String moveType = obj["type"].as<String>();
+    if (moveType == "freedrive") {
+      currentMove.moveType = eFreeDrive;
+      currentMove.params.freedriveParams.direction = obj["direction"];
+      currentMove.params.freedriveParams.duration = obj["duration"];
+    } else if (moveType == "linefollow") {
+      currentMove.moveType = eLineFollow;
+      currentMove.params.linefollowParams.stopDistance = obj["stopDistance"];
+      currentMove.params.linefollowParams.speed = obj["speed"];
+    } else if (moveType == "scissor") {
+      currentMove.moveType = eScissor;
+      currentMove.params.scissorParams.direction = obj["direction"];
+    } else if (moveType == "belt") {
+      currentMove.moveType = eBelt;
+      currentMove.params.beltParams.direction = obj["direction"];
+      currentMove.params.beltParams.duration = obj["duration"];
+    } else if (moveType == "calibrate") {
+      currentMove.moveType = eCalibrate;
+      currentMove.params.calibrationParams.duration = obj["duration"];
+    } else {
+      // If move unknown,
+      continue;  // Skip this move
+    }
```

Comment:
```suggestion
    auto moveType = obj["type"].as<MoveType>()
    switch(moveType) {
        case eFreedrive:
        currentMove.moveType = moveType;
        currentMove.params.freedriveParams.direction = obj["direction"];
        currentMove.params.freedriveParams.duration = obj["duration"];
        break;
    }
```

I'm not going to write all of this out because I think you can see the idea.
This does mean the moveType will be stored as the number in Json, but this is definitely preferable to all the String comparisons, and if elses

Reply:
This is definitely preferable to the strcmps and unnecessary bytes. 

Lets take a look at doing this next time we are actually with the robot. I want to do it but I don't want to risk breaking something without being able to test. 

As long as we have a lookup table (.txt in our repo, even) the ints in the json should be fine.